nsm/main.go

```diff
@@ -0,0 +1,37 @@
+package main
+
+import (
+	"github.com/fsnotify/fsnotify"
+	pluginapi "k8s.io/kubernetes/pkg/kubelet/apis/deviceplugin/v1beta1"
+	"log"
+	"os"
+)
+
+func main() {
+	log.Println("Loading NSM")
+	log.Println("Starting FS watcher.")
+	log.Println(pluginapi.DevicePluginPath)
+	watcher, err := newFSWatcher(pluginapi.DevicePluginPath)
+	if err != nil {
+		log.Println("Failed to created FS watcher.")
+		os.Exit(1)
+	}
+	defer watcher.Close()
```

Comment:
Does newFSWatcher block? If not, this function will exit, close the connection, and kill the application.

Take a look at https://golang.org/pkg/sync/#WaitGroup

```
var wg sync.WaitGroup
wg.Add(1)
go run(wg)
wg.Wait()
```

run calls ```wg.Done()``` when it is complete.